web/src/upload.js

```diff
@@ -302,6 +302,48 @@ export function setup_upload(config) {
     $drag_drop_container.on("dragover", (event) => event.preventDefault());
     $drag_drop_container.on("dragenter", (event) => event.preventDefault());
 
+    $("#main_div").on("dragover", (event) => event.preventDefault());
+    $("#main_div").on("dragenter", (event) => event.preventDefault());
+
+    $("#main_div").on("drop", (event) => {
+        event.preventDefault();
+
+        // Check for edit box which registered the event handler no longer exists.
+        if (!$drag_drop_container.closest("html").length) {
+            return;
+        }
+
+        const $drag_drop_edit_containers = $(".message_edit_form form");
+        for (const container of $drag_drop_edit_containers) {
+            // This means file was dropped on #compose, which will be handled by different function
+            if (container.contains(event.target)) {
+                return;
+            }
+        }
```

Comment:
I theorize that this block isn't necessary, and should instead by implemented via the `drag_drop_container.on("drop", (event) => {` block just below having an `e.stopPropagation` -- that should ensure that dragging into compose/edit windows don't call this function without any special logic here.